### Importing Preprocessed Dataset

In [7]:
import numpy as np
import pandas as pd
import seaborn as sns

import warnings
warnings.filterwarnings('ignore')

In [8]:
# import dataset
df = pd.read_csv('preprocessed_housePrice.csv')

In [9]:
df.head()

price  resid_area  air_qual  room_num   age  teachers  poor_prop  \
0   24.0       32.31     0.538     6.575  65.2      24.7       4.98   
1   21.6       37.07     0.469     6.421  78.9      22.2       9.14   
2   34.7       37.07     0.469     7.185  61.1      22.2       4.03   
3   33.4       32.18     0.458     6.998  45.8      21.3       2.94   
4   36.2       32.18     0.458     7.147  54.2      21.3       5.33   

   n_hos_beds  n_hot_rooms  rainfall     parks  Sold  avg_dist  airport_YES  \
0       5.480     11.19200        23  0.049347     0    4.0875            1   
1       7.332     12.17280        42  0.046146     1    4.9675            0   
2       7.394     46.19856        38  0.045764     0    4.9675            0   
3       9.268     11.26720        45  0.047151     0    6.0650            1   
4       8.824     11.28960        55  0.039474     0    6.0625            0   

   waterbody_Lake  waterbody_Lake and River  waterbody_River  
0               0                         0                1  
1               1                         0                0  
2               0                         0                0  
3               1                         0                0  
4               1                         0                0

### Simple Logistic Regression

In [10]:
# single Independent Variable
x = df[['price']]

In [12]:
x.head()

price
0   24.0
1   21.6
2   34.7
3   33.4
4   36.2

In [11]:
# Dependent variable
y = df['Sold']

In [13]:
y.head()

0    0
1    1
2    0
3    0
4    0
Name: Sold, dtype: int64

**Simple Logistic Regression using sklearn library**

In [14]:
# importing Logistic regression from sklearn
from sklearn.linear_model import LogisticRegression

In [15]:
# creating object
slr = LogisticRegression()

In [16]:
# training model
slr.fit(x, y)

LogisticRegression()

In [18]:
# coefficent and intercept
print("coefficent[beta1] of model: ", slr.coef_)
print("intercept[beta0] of model: ", slr.intercept_)

coefficent[beta1] of model:  [[-0.03571865]]
intercept[beta0] of model:  [0.61477516]


**Simple Logistic Regression using statsmodels library**

In [19]:
# importing library
import statsmodels.api as sm

In [20]:
x_constant = sm.add_constant(x)

In [21]:
x_constant.head()

const  price
0    1.0   24.0
1    1.0   21.6
2    1.0   34.7
3    1.0   33.4
4    1.0   36.2

In [22]:
import statsmodels.discrete.discrete_model as sd

In [23]:
sm_slr = sd.Logit(y, x_constant).fit()

Optimization terminated successfully.
         Current function value: 0.676690
         Iterations 5


In [24]:
sm_slr.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                           Logit Regression Results                           
==============================================================================
Dep. Variable:                   Sold   No. Observations:                  506
Model:                          Logit   Df Residuals:                      504
Method:                           MLE   Df Model:                            1
Date:                Wed, 13 Jan 2021   Pseudo R-squ.:                 0.01788
Time:                        15:58:39   Log-Likelihood:                -342.41
converged:                       True   LL-Null:                       -348.64
Covariance Type:            nonrobust   LLR p-value:                 0.0004142
==============================================================================
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
const          0.6149      0.248      2.484      0.013       0.130       1.100
price         -0.0357      0.010     -3.417      0.001      -0.056      -0.015
==============================================================================
"""

- Coef of constant [beta 0] = 0.6149
- Coef of price [beta 1] = -0.0357
- we have z values for coef
- Our method is Maximum Likely Hood [MLE]